src/cpp/core/include/core/http/Response.hpp

```diff
@@ -320,10 +323,13 @@ class Response : public Message
       }
       
       // set content type
-      setContentType(filePath.getMimeContentType());
+      if (contentType().empty())
+      {
+         setContentType(filePath.getMimeContentType());
```

Comment:
I'm hoping we don't inadvertently rely on the former unconditional statement somewhere else in the code -- as in the Content-Type header is set even though it shouldn't be, but it's been masked by the former logic.

We'll have to keep our eyes peeled.